### PR TITLE
Hash Table implementation using CRC32 algorithm

### LINKED ISSUE #1289: Implementing Hashtables using  CRC-32 algorithm.
### Brief Summary: 
Hashtable is a data structure that implements an associative array abstract data type, a structure that can map keys to values. This article explains the implementation of hash tables using the CRC-32 algorithm.
### Key Takeaways: 

1. Creating a hash table.
2. Performing Various Operations on Hashtable 
2. Hash function used for implementation.
3. Collision detection in hash tables

### References: 
N/A

### PR DESCRIPTION
**Pull request by jerimkaura**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1289

### CODE REVIEW

File: articles/hashtables-implementation-using-CRC32-algorithm/index.md [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+        #random integer for use when the key is not a string
+        self.a: int = randint(1, 2**32)
+
+        # random integer for use in secondary hashing
+        self.b: int = randint(1, 2**32)
+```
+
+### 2. Class methods & hashing
+
+We have successfully created the data members. We now proceed to write our dunder methods for overwriting some of the Python built-in function behaviors. This implementation is based on the following class methods, which will be discussed after the dunder methods.
```
Comment (lalith1403) [inline articles/hashtables-implementation-using-CRC32-algorithm/index.md:51]: Explicitly mention double under methods are also known as dunder methods.

File: articles/hashtables-implementation-using-CRC32-algorithm/index.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+
+```python
+def __len__(self) -> int:
```
Comment (lalith1403) [inline articles/hashtables-implementation-using-CRC32-algorithm/index.md:54]: Check the indentation. I think a tab should right indent this.

File: articles/hashtables-implementation-using-CRC32-algorithm/index.md [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+We then define our `HashTable` class with its data members and methods.
+
+```python
+class HashTable:
+    def __init__(self):
+        # Initial table size
+        self.table_size: int = 23
+
+        # initilizing all table slots to none
+        self.table: List[(T, T)] = [None] * self.table_size
```
Comment (lalith1403) [inline articles/hashtables-implementation-using-CRC32-algorithm/index.md:25]: The variable T is not defined. Can you check the entire code once? I am getting errors
1. Indentation errors
2. Name Errors. 

Add a repl link which will ensure that we know the code is working. For example:

Run this code [here](https://repl.it/@lalithNarayan/SuperiorDeficientType).

File: articles/hashtables-implementation-using-CRC32-algorithm/index.md [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+We have successfully created the data members. We now proceed to write our dunder methods for overwriting some of the Python built-in function behaviors. This implementation is based on the following class methods, which will be discussed after the dunder methods.
+
+```python
+def __len__(self) -> int:
+        """ Returns number of (key, value) pairs in table """
+        return self.filled_count
+
+    def __repr__(self) -> str:
+        """ Returns HashTable's string representation
+        (à la Python's dict's {key1: value1, key2: value2, ..., keyN: valueN})
```
Comment (lalith1403) [inline articles/hashtables-implementation-using-CRC32-algorithm/index.md:60]: What is this? **à la** in à la Python's dict's?

**Review by lalith1403 — CHANGES_REQUESTED** [review]
It is not clear which are class methods and which are actual functions. Also, you need to create an object of the class and run it. It would help if you explain every single step in detail. I see each section is disconnected from each other. 

Also, paste the expected output from the terminal. Not the _passed the tests_ output, but something that explains the concept.

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**